### PR TITLE
Add UI to create and delete api keys

### PR DESCRIPTION
**Pull request by cquiroz**
Adds a user preferences UI where we can see some user properties and manage api keys

image

### CODE REVIEW

File: common-graphql/src/main/scala/queries/schema/SSO.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+package queries.schemas
+
+import clue.annotation.GraphQLSchema
+
+@GraphQLSchema
+trait SSO:
+  object Scalars:
+    type UserId   = String
+    type RoleId   = String
```
Comment (rpiaggio) [inline common-graphql/src/main/scala/queries/schema/SSO.scala:12]: Can use use `User.Id` and `StandardRole.Id` here?

File: common/src/main/scala/explore/model/GraphQLClients.scala [hunk 2]
```diff
@@ -9,31 +9,39 @@ import cats.syntax.all.*
...
+import org.http4s.syntax.all.*
 import org.typelevel.log4cats.Logger
 import queries.schemas.*
 
 case class GraphQLClients[F[_]: Async: Parallel] protected (
   odb:           WebSocketClient[F, ObservationDB],
   preferencesDB: WebSocketClient[F, UserPreferencesDB],
-  itc:           TransactionalClient[F, ITC]
+  itc:           TransactionalClient[F, ITC],
+  sso:           WebSocketClient[F, SSO]
```
Comment (rpiaggio) [inline common/src/main/scala/explore/model/GraphQLClients.scala:23]: Is it worth it to keep a web socket connection open to SSO in every instance of explore? Maybe a `TransactionalClient` is enough?

(Or are you using it temporarily while we cannot refresh the token in a TransactionalClient?)

Comment (cquiroz) [reply common/src/main/scala/explore/model/GraphQLClients.scala:23]: Yes I want to use `TransactionalClient` but we need to be able to set the auth token first

File: common-graphql/src/main/scala/queries/schema/SSO.scala (hunk 1 above)
Comment (cquiroz) [reply common-graphql/src/main/scala/queries/schema/SSO.scala:12]: Let me try that

**Review by rpiaggio — APPROVED** [review]
LGTM

**Pull request merged**